swap_meet/vendor.py

```diff
@@ -1,2 +1,64 @@
+from operator import attrgetter
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        if inventory == None:
+            self.inventory = []
+        else:
+            self.inventory = inventory 
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+        
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False 
+    
+    def get_by_category(self, category):
+        same_catagory_items = []
+        for item in self.inventory:
+            if item.category == category:
+                same_catagory_items.append(item)
+        return same_catagory_items
+
+
+    def swap_items(self, friend_vendor, my_item, their_item):
+        if my_item not in self.inventory:
+            return False
+        elif their_item not in friend_vendor.inventory:
+            return False
+        else:
+            self.remove(my_item)
+            self.add(their_item)
+            friend_vendor.remove(their_item)
+            friend_vendor.add(my_item)
+        return self.inventory 
+
+    def swap_first_item(self, friend_vendor):
+        if self.inventory == [] or friend_vendor.inventory == []:
+            return False
+        else:
+            self.swap_items(friend_vendor, self.inventory[0], friend_vendor.inventory[0])
+            return True
+
+    def get_best_by_category(self, category):
+        similiar_items = []
+        for item in self.inventory:
+            if item.category == category:
+                similiar_items.append(item)
+        if len(similiar_items) == 0:
+            return None
+        best_condition_item = max(similiar_items, key=attrgetter('condition'))
+        return best_condition_item
```

Comment:
Nice logic here! Do be careful to make sure you know what `attrgetter` is doing if you're using it. Alternatively, can you think of how you could write the key using a lambda instead of `attrgetter`?